addons/controllers/clusterbootstrap_controller.go

```diff
@@ -196,7 +197,8 @@ func (r *ClusterBootstrapReconciler) reconcileNormal(cluster *clusterapiv1beta1.
 func (r *ClusterBootstrapReconciler) createOrPatchclusterBootstrapFromTemplate(cluster *clusterapiv1beta1.Cluster,
 	log logr.Logger) (*runtanzuv1alpha3.ClusterBootstrap, error) {
 
-	tkrName := util.GetTKRNameForCluster(r.context, r.Client, cluster)
+	// TODO: What about those classy clusters that aren't labeled with any TKR related labels?
+	tkrName := util.GetTKRNameForCluster(r.context, r.Client, cluster, constants.TKRLabelClassyClusters)
```

Comment:
We should already know the TKR name here. So, no need to get and check. We could just pass it as an argument to createOrPatchclusterBootstrapFromTemplate().